episodes/2-keras.Rmd

```diff
@@ -735,17 +740,16 @@ visualize a confusion matrix.
 
 :::: solution
 ## Solution
-The confusion matrix shows that the predictions for Adelie and Gentoo
-are decent, but could be improved. However, Chinstrap is not predicted
-ever.
-The training loss was very low, so from that perspective this may be
-surprising.
-But this illustrates very well why a test set is important when training
-neural networks.
-We can try many things to improve the performance from here.
-One of the first things we can try is to balance the dataset better.
-Other options include: changing the network architecture or changing the
-training parameters
+The confusion matrix shows that the predictions for Adelie and Gentoo are decent, but could be improved. However, Chinstrap is not predicted ever.
+Why the chinstraps are not present in the the predictions in the confusion matrix?
+If we go back to the [**Pair Plot**](https://carpentries-incubator.github.io/deep-learning-intro/2-keras.html#pair-plot) in the Visualization section, we can figure it out that the biggest challenge is masking the Chinstrap penguins from the marginal distributions of the four features (bill length, bill depth, flipper length, and body mass). That is, There is no single variable that separates Chinstrap from all other species. The combination of bill_length_mm vs. bill_depth_mm (the 2nd subfigure in the first row in the pairplot) gives a good separation of Chinstrap from Adelie and Gentoo.
```

Comment:
I think this should be a relative link to make it work in case servers or paths change. I *think* this should work:
```suggestion
If we go back to the [**Pair Plot**](#pair-plot) in the Visualization section, we can figure it out that the biggest challenge is masking the Chinstrap penguins from the marginal distributions of the four features (bill length, bill depth, flipper length, and body mass). That is, There is no single variable that separates Chinstrap from all other species. The combination of bill_length_mm vs. bill_depth_mm (the 2nd subfigure in the first row in the pairplot) gives a good separation of Chinstrap from Adelie and Gentoo.
```
But the link needs to be tested.